WikiClientLibrary/Generators/RandomPageGenerator.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using WikiClientLibrary.Client;
+using WikiClientLibrary.Generators;
+using WikiClientLibrary.Generators.Primitive;
+using WikiClientLibrary.Pages;
+using WikiClientLibrary.Sites;
+
+namespace WikiClientLibrary.Generators
+{
+    /// <summary>
+    /// Gets random pages in a specific namespace.
+    /// </summary>
+    class RandomPageGenerator : WikiPageGenerator
+    {
+        /// <inheritdoc />
+        public RandomPageGenerator(WikiSite site) : base(site)
+        {
+        }
+
+        /// <inheritdoc />
+        protected override WikiPageStub ItemFromJson(JToken json)
+        {
+            return new WikiPageStub((int)json["id"], (string)json["title"], (int)json["ns"]);
+        }
+
+        /// <inheritdoc />
+        public int NamespaceId { get; set; } = 0;
```

Comment:
From [mw:API:Random](https://www.mediawiki.org/wiki/API:Random), `NamespaceId` should be its plural form `NamespaceIds`, and accepts a set of namespace IDs
> `rnnamespace`: Only list pages in these namespaces. (Default: all namespaces)

You may take a look at [`BacklinksGenerator.NamespaceIds`](https://github.com/aquilla12/WikiClientLibrary/blob/51213618783993ffadbc91f44e79d528ca68bd62/WikiClientLibrary/Generators/BacklinksGenerator.cs#L43) for reference on implementation.